outlines/caching.py

```diff
@@ -7,7 +7,10 @@
 import cloudpickle
 from diskcache import Cache
 
-_caching_enabled = True
+if os.environ.get("OUTLINES_DISABLE_CACHE", "") in ("", "0"):
```

Comment:
Why do we need an environment variable for this?  

Also, can't we use the existing `disable_cache` function?

Reply:
It may be beneficial to introduce an environment variable that allows users disable cache, even outside of pytest.

I could also simply make the `disable_cache` fixure patch this variable and have no changes to `caching.py` if we don't want to expose this.